### ISSUE TITLE
MySQL zero fill makes eager loading impossible

### ISSUE THREAD

**Issue by JapSeyz** [laravel/framework#53500]
### Laravel Version

v10.48.23

### PHP Version

8.3.13

### Database Driver & Version

11.5.2-MariaDB-ubu2204-log

### Description

We have an ID column defined as `bigint(13) unsigned zerofill` because it stores 13-digit EAN barcodes.

When eager loading this table, it provides the follow, fully functional query:

```sql
select * from `table`.`barcodes` where `table`.`barcodes`.`id` in (978001, 978002, 978003)
```

I have validated that this fetches the correct items, but they are somehow never mapped onto the model they're being eagerloaded onto. Presumably because ->id is gonna be prefixed by `0`s to pad it to 13 digits, which then makes the `barcode_id` on the parent-model != `id` on the eagerloaded model

### Steps To Reproduce

Try to eager load a model where the id column is `zerofill` and the id is less than what it will be zero filled to.

**Comment by iMusicJJ** [comment]
No, but I doubt that would solve it. That wouldn't remove the extra zero filled `0`s, right?

Edit: I tried it, and it does _not_ work.

**Comment by JapSeyz** [comment]
No, but I doubt that would solve it. That wouldn't remove the extra zero filled 0s, right?

Edit: I tried it, and it does not work.

**Comment by JapSeyz** [comment]
I know, but data is returned prefixed by `0`s to the defined length. in this case 13.

It means when we look up `id = 9876`, the id returned will be `0000000009876` (which is correct). The id will then no longer match, when laravel tries to attach the model to the parent.

The issue is not within the database, but how Laravel handles zero fill. I am not sure it is a bug, but it was something that took a while to track down, so creating an issue for it, at least might help someone find it in the future.

I am not sure what your comment is trying to say.
